### ISSUE TITLE
Breaking when PODs have HTTP GET readiness and liveness probes and they do not start with a /

### ISSUE THREAD

**Issue by damadei-google** [istio/istio#27583]
**Bug description**
We noticed that when a POD has an HTTP GET request path that does not start with an /, we get an HTTP 500 error

[ ] Docs
[ ] Installation
[x ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Health check rewrite should work with urls starting with / or not

**Steps to reproduce the bug**
Create a deployment, specify the path of the HTTP GET health check to start without a /, see that the PODs does not come up due to readiness failure

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used 
Helm)**
client version: 1.6.8-asm.9
control plane version: 1.6.8-asm.9
data plane version: 1.6.8-asm.9 (4 proxies)

Client Version: v1.16.13-dispatcher
Server Version: v1.16.13-gke.401

**How was Istio installed?**
Via Anthos Service Mesh installer on GKE on GCP

**Environment where bug was observed (cloud vendor, OS, etc)**
GKE with Container OS

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching the dump file to this issue.

**Comment by esnible** [comment]
I tried to reproduce this and failed.  I deployed Istio's _httpbin_ sample, with the following additional probe:

```
        # See https://istio.io/latest/docs/ops/configuration/mesh/app-health-check/
        readinessProbe:
          httpGet:
            path: status/200
            port: 80
          initialDelaySeconds: 5
          periodSeconds: 5
```

Note the missing `/` in the path.  I verified that the pod became active, and I verified the injected probe worked.
While testing this, I did get 500 errors, but they occurred when I used the wrong port, 8000 instead of 80.  The behavior changes with Istio ... without Istio, the probe was reporting "connection refused", with Istio it was reporting HTTP 500 status.

Can you tell us what your readinessProbe httpGet values are?  Also, can you tell us what you see when you do

- `kubectl exec deploy/httpbin -c istio-proxy -- curl -v http://localhost:15020/app-health/httpbin/readyz` (replace httpbin with your container name)
- `kubectl exec deploy/httpbin -c istio-proxy -- curl -v http://localhost:80/status/200` (replace 80 with your port and "status/200" with your path")

**Comment by damadei-google** [comment]
Hi Ed,

We were able to simulate in 2 different scenarios and in my customer's case it was with Spring Boot + Actuator. I'll try to create a small reproducer and get back to you.

Daniel

**Comment by damadei-google** [comment]
@esnible,

Please try this one:

apiVersion: apps/v1
kind: Deployment
metadata:
  name: repro
  namespace: repro
  labels:
    name: repro
spec:
  replicas: 1
  selector:
    matchLabels:
      name: repro
  template:
    metadata:
      labels:
        name: repro
    spec:
      containers:
        - name: repro
          image: damadei/repro:latest
          imagePullPolicy: Always
          ports:
            - containerPort: 8080
          readinessProbe:
            failureThreshold: 3
            httpGet:
              path: actuator/health
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 10
            periodSeconds: 10
            successThreshold: 1
            timeoutSeconds: 1
          livenessProbe:
            failureThreshold: 3
            httpGet:
              path: actuator/health
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 10
            periodSeconds: 10
            successThreshold: 1
            timeoutSeconds: 1

Then change to /actuator/health and see it will work.

**Comment by esnible** [comment]
Your deployment worked for me under Istio without changing the readiness URL.  I am using Istio master (pre-1.8), Kubernetes 1.17.

```
kubectl create ns repro
kubectl apply -f https://github.com/istio/istio/files/5295742/06-deployment.yaml.txt
kubectl -n repro get pods # it becomes ready, 1/1
kubectl delete -f https://github.com/istio/istio/files/5295742/06-deployment.yaml.txt
kubectl label ns repro istio-injection=enabled
kubectl apply -f https://github.com/istio/istio/files/5295742/06-deployment.yaml.txt
kubectl -n repro get pods # it becomes ready, 2/2
```

**Comment by damadei-google** [comment]
Weird... I'm using Istio 1.6.8 in Anthos Service Mesh install.

**Comment by howardjohn** [comment]
Ah I think this was fixed in #24412, but not older versions

**Comment by howardjohn** [comment]
Closing this as this is fixed by https://github.com/istio/istio/pull/24412. due to the impending EOL of 1.6 and the ease of workaround we are unlikely to backport
